src/backend/upload_file/upload_file_for_chunks.py

```diff
@@ -0,0 +1,172 @@
+logger = logging.getLogger(__name__)
+
+class CheckFileType:
+
+    def check_file_type(self, filename):
+        mime_to_extension = {
+            "application/pdf": ".pdf"
+        }
+        try:
+            mime = magic.Magic(mime=True)
+            file_type = mime.from_file(filename)
+            extension = mime_to_extension.get(file_type)
+            if not extension:
+                logger.error(f"Unknown MIME type detected: {file_type}")
+                return None
+            return extension
+        except Exception as e:
+            logger.error(f"Error detecting file type for {filename}: {e}")
+            return None
+
+
+class ExtractFiles:
+
+    def __init__(self):
+
+        # Initialize EasyOCR Reader
+        self.ocr_reader = easyocr.Reader(['en'], gpu=False)
+
+    def format_table_as_text(self, table):
+        table = [[str(item) if item is not None else "" for item in row] for row in table]
+        col_widths = [max(len(str(item)) for item in col) for col in zip(*table)]
+        row_format = " | ".join(f"{{:<{w}}}" for w in col_widths)
+        formatted_table = "\n".join(row_format.format(*row) for row in table)
+        return formatted_table.strip()
+
+    def extract_pdf_with_layout(self, filename, output_json_path):
+        results = []
+
+        with pdfplumber.open(filename) as pdf:
+            for page_number, page in enumerate(pdf.pages, start=1):
+                page_elements = []
+
+                for block in page.extract_words():
+                    page_elements.append({
+                        "type": "text",
+                        "content": block["text"],
+                        "x0": block["x0"],
+                        "y0": block["top"]
+                    })
+
+                for table in page.extract_tables():
+                    table_text = self.format_table_as_text(table)
+                    bbox = page.find_tables()[0].bbox  # Get bounding box for the first table
+                    page_elements.append({
+                        "type": "table",
+                        "content": table_text,
+                        "x0": bbox[0],
+                        "y0": bbox[1]
+                    })
+
+                    pil_image = page.to_image(resolution=300).original
+                    numpy_image = np.array(pil_image)
+                    ocr_result = self.ocr_reader.readtext(numpy_image)
+                    ocr_text = " ".join([text[1] for text in ocr_result])
+
+                    page_elements.append({
+                        "type": "image",
+                        "content": f"Data obtained from image: {ocr_text.strip()}",
+                        "x0": 0,  # Assuming full-page images
+                        "y0": 0
+                    })
+
+                page_elements.sort(key=lambda e: (e["y0"], e["x0"]))
+
+                page_text = " ".join([element["content"] for element in page_elements])
+
+                results.append({
```

Comment:
Can you put page number into metadata uber field

                results.append({
                    metadata: {"page_number": page_number},
                    "text": page_text
                })